### ISSUE TITLE
issue with retrieving CRS using `parse_crs`

### ISSUE THREAD

**Issue by maboualidev** [laspy/laspy#230]
When trying to retrieve CRS information, the following method does not work:

```Python
las_fid = laspy.open('USGS_LPC_DE_DelawareValley_HD_2015_18TWK517443_LAS_2017.laz')
crs = las_fid.header.parse_crs()  # returns None
print(type(crs))  # Prints NoneType
```

However, QGIS has no issue retrieving the CRS; hence, those information are available somewhere, which encouraged me to dig more.

Digging more:

Trying accessing vlrs:
```Python
las_fid.header.vlrs
```
will give:
```bash
[
    <VLR(user_id: 'NIIRS10', record_id: '4', data len: 10)>, 
    <VLR(user_id: 'NIIRS10', record_id: '1', data len: 26)>
]
```

trying accessing `evlrs` through `las_fid` or `las_fid.header` I get the error that such attributes does not exists.

HOWEVER, if I do

```Python
data = las_fid.read()
```
still `data.header.parse_crs()` returns `None` BUT

```Python
data.evlrs
```
returns the following:

```bash
[
    <laspy.vlrs.known.WktCoordinateSystemVlr object at 0x183e2cb50>, 
    <VLR(user_id: 'LASF_Spec', record_id: '7', data len: 569)>
]
```

and

```Python
data.evlrs.get('WktCoordinateSystemVlr')[0].parse_crs()
```
properly returns a CRS object

So to summarize:
- parse_crs does not return proper CRS
- evlrs seems to be un-available when doing `laspy.open`
- evlrs becomes available after doing full read (but it is not available through header but directly from data itself
    - `data.heaer.evlrs` doesn't exists and I get attributes does not exists error.
- only after doing full read evlrs becomes available and we can manually extract the crs but not through `parse_crs`

laspy Version:
```bash
laspy                     2.2.0              pyh8a188c0_0    conda-forge
```

Download URL for USGS LAZ file to reproduce the issue:
USGS LAZ File URL to download: [USGS_LPC_DE_DelawareValley_HD_2015_18TWK517443_LAS_2017](https://rockyweb.usgs.gov/vdelivery/Datasets/Staged/Elevation/LPC/Projects/USGS_LPC_DE_DelawareValley_HD_2015_LAS_2017/laz/USGS_LPC_DE_DelawareValley_HD_2015_18TWK517443_LAS_2017.laz)

**Comment by ssfabiopicchi** [comment]
I opened a PR in an attempt to fix that issue:
https://github.com/laspy/laspy/pull/227

I didn't know the CRS values were stored in evlrs. Is that valid? I believe the spec status that it should be in vlrs.

Before parse_crs, this is how I was getting it:
```
crs = pyproj.CRS.from_wkt(las_file.header.vlrs.get("WktCoordinateSystemVlr"))[0].string)
```

**Comment by hobu** [comment]
> I didn't know the CRS values were stored in evlrs. Is that valid? 

Yes. In LAS 1.4, it is a possible and common practice to use EVLRs to update or override SRS information that was written in a VLR.

> I believe the spec status that it should be in vlrs.

For LAS versions 1.0-1.3, this is true. For 1.4, the SRS information can be overridden with an EVLR.

**Comment by tmontaigu** [comment]
The PR #227 won't fix this issue, as the problem in this case is that the CRS VLR is stored as an EVLRs.

We currently don't read evlrs, unless the user uses `laspy.read`.
#227 fixes the case when there are 2 CRS VLR and only one of them actually containts the information.

#232 will make it so that EVLRs are read when the file is opened, so doing this would work:
```Python
with laspy.open(file):
   file.evlrs.get('WktCoordinateSystemVlr')[0].parse_crs()
``` 
However I can't just put the EVLRs read into the `header.vlrs`
as it would break too many things. This means that `data.header.parse_crs` still won't work.

I'm going to think a bit about how to make it work, I see 2 options:
- Adding a `LasData.parse_crs` and `LasReader.parse_crs` and these methods would take care of trying both the `header.parse_crs` and look in into the evlrs.
- Moving the `.evlrs` member from `LasData` / `LasReader` into `LasHeader` which would be a bit weird and kinda break the current API

**Comment by maboualidev** [comment]
Hi,

I am currently doing something like this to get past this:

```python
import laspy
from pyproj import CRS
from typing import Optional, Tuple


def parse_crs(las_fid: laspy.LasReader) -> Tuple[CRS, Optional[laspy.LasData]]:
    points = None
    crs = las_fid.header.parse_crs()
    
    if crs is None:
        # Checking if crs is in evlrs
        points = las_fid.read()
        if hasattr(points, 'evlrs'):
            crs_info_evlrs = []
            for evlrs_str in ('WktCoordinateSystemVlr', 'GeoKeyDirectoryVlr'):
                crs_info_evlrs.extend(points.evlrs.get(evlrs_str))

            if len(crs_info_evlrs) > 0:
                crs = crs_info_evlrs[0].parse_crs()

                
    # crs should have been resolved by now
    if crs is None:
            raise ValueError("Could not find CRS information.")

    return crs, points
```

I can make a PR if you are interested in including this. (I don't recommend it though since this is changing the output type of `parse_crs`. So, this would break a lot of people's code who rely on the current version of `parse_crs`)

The issue with this version is that it loads all the data. So, if you are relying on loading the data in chunks, this is going to load all the data.

@tmontaigu changes in #232 should fix the problem of not requiring to load all the data. In that case, we can change the above code to:

```Python
def parse_crs(las_fid: laspy.LasReader) -> CRS:
    crs = las_fid.header.parse_crs()
    
    if crs is None:
        # Checking if crs is in evlrs
        if hasattr(las_fid, 'evlrs'):
            crs_info_evlrs = []
            for evlrs_str in ('WktCoordinateSystemVlr', 'GeoKeyDirectoryVlr'):
                crs_info_evlrs.extend(las_fid.evlrs.get(evlrs_str))

            if len(crs_info_evlrs) > 0:
                crs = crs_info_evlrs[0].parse_crs()

    # crs should have been resolved by now
    if crs is None:
            raise ValueError("Could not find CRS information.")

    return crs
```

**Comment by tmontaigu** [comment]
For now i'm just going to think about what would be the best, I'd prefer take a bit of time to add something
than rushing and having to deprecate it afterwards
